tuftool/src/create.rs

```diff
@@ -239,13 +240,23 @@ impl<'a> CreateProcess<'a> {
             .to_owned();
 
         let mut file = File::open(path).context(error::FileOpen { path })?;
-        let mut digest = Sha256::new();
-        let length = std::io::copy(&mut file, &mut digest).context(error::FileRead { path })?;
+        let mut digest = Context::new(&SHA256);
+        let mut buf = [0; 8 * 1024];
+        let mut length = 0;
+        loop {
+            match file.read(&mut buf).context(error::FileRead { path })? {
+                0 => break,
+                n => {
+                    digest.update(&buf[..n]);
+                    length += n as u64;
+                }
+            }
+        }
```

Comment:
I'm not familiar with `ring` but I'm assuming from context this loop is required because copy() doesn't play nice with `digest` appearing as a slice?

Reply:
It's necessary because `ring::digest::Context` doesn't implement `std::io::Write` (although it easily could).